### ISSUE TITLE
Enable  hover hints  runtime var  (value)  like pull 9191  (Maybe Ctrl + mouse hover)

### ISSUE THREAD

**Issue by pythonwood** [spyder-ide/spyder#13312]
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

### What steps reproduce the problem?

1. 
2. 
3. 

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 
* Python version: 
* Qt version: 
* PyQt version: 
* Operating System name/version: 

### Dependencies

```
PASTE DEPENDENCIES HERE
```

Enable  hover hints  runtime var  (value)  like pull 9191 
(Maybe Use Ctrl + mouse hover)

It is Great doc feature.  
PR: Enable hover hints #9191
https://github.com/spyder-ide/spyder/pull/9191

**Comment by dalthviz** [comment]
Hi @pythonwood thanks for the feedback! I'm unsure what do you mean by this, could you elaborate a little bit more? Maybe an step by step example using a code snippet could help :)

**Comment by pythonwood** [comment]
sometime there ara many variable in the variable manager windows that I use more than 8 seconds to  find the varible and get the runtime value.


and I had  searched and found   [9191](https://github.com/spyder-ide/spyder/pull/9191) 

so My idea is : 
 (when hover hint a word in source)
if it is a runtime time **variable**:  then show the  value of it
if it is a **class/fuction**:  then show the doc

**Comment by dalthviz** [comment]
Thanks @pythonwood for the clarification! @spyder-ide/core-developers what do you guys think?

**Comment by jitseniesen** [comment]
I think it will be useful. However, I also think we have to be careful about scope which makes it not straightforward to implement this idea.

**Comment by dalthviz** [comment]
Yep that's what I was thinking too, thanks @jitseniesen for the feedback! :+1:

**Comment by CAM-Gerlach** [comment]
To be pedantic. in in Python parlance, (per the [official reference](https://docs.python.org/3/reference/), there isn't a strictly defined concept for what a "variable" is per say, as there is in C/++. Everything (including functions, methods, classes, builtin types, instances, etc) is an _object_ in memory, and this is distinct from the _name(s)_ for those objects (which is what is directly shown in Spyder's Editor), and there is nothing at the language level preventing any name from rebound to another object.

Pedantics aside, practically speaking we could do a simple lookup for the hovered-over name in the namespace of the currently active console, and retrieve and display its value or a summary thereof if it matched some heuristic (for example, the current Variable Explorer filters).

However, for this to work, the name would need to be defined in the top-level namespace and main-level scope and in many cases, e.g. arrays, dataframes, collections, etc. the name does appear in the variable explorer by default but the values able to be displayed in a calltip are rather limited, while in many cases the user might want the the help hover hint instead (which provides useful details about the class, its attributes, its constructor, etc). So I'm not sure what should do in these classes—display both? And the popup could become misleading if the name being hovered over is in a different scope but is the same as one defined globally or interactively. So we'd have to think carefully about how to implement this.
